api/src/main/java/org/pmiops/workbench/api/CloudTaskUserController.java

```diff
@@ -95,4 +104,40 @@
     log.info(String.format("successfully audited %d users", request.getUserIds().size()));
     return ResponseEntity.noContent().build();
   }
+
+  @Override
+  public ResponseEntity<Void> synchronizeUserAccess(SynchronizeUserAccessRequest request) {
+    int errorCount = 0;
+    for (long userId : request.getUserIds()) {
+      DbUser user = userDao.findUserByUserId(userId);
+
+      try {
+        // 2FA synchronization requires an outgoing call to gsuite. For this reason, we
+        // optimize to only verify that users who have 2FA enabled, still have it enabled.
+        // Users who don't have 2FA will go through an active flow to enable it, and are not
+        // dependent on this offline check. Disabled users have no access anyways, so don't
+        // bother checking them either.
+        if (user.getTwoFactorAuthCompletionTime() != null && !user.getDisabled()) {
```

Comment:
Thanks, I forgot about these changes. I'd prefer to explicitly check completed here; I added one way of exposing this information which mirrors another API provided by that service, PTAL.

The reason I prefer this over `isCompliant` is that there is no reason to check the 2FA status for a bypassed user. It's not critical for 2FA though, since we also should never be bypassing anyone for 2FA in higher environments. Mostly this will give some savings in test/perf.

Reply:
Sorry, I am bit confused this part: "there is no reason to check the 2FA status for a bypassed user".
If that is true, then isCompliant is the right method to use. 
It returns true if use bypassed such module: https://github.com/all-of-us/workbench/blob/master/api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java#L130
Then we can skip that check.
The method you added just check completion timestamp but ignore bypass status.

Reply:
I don't really want bypasses to have any effect here.

Here is my desired logic:

`!completed, !bypassed` - no check
`completed, !bypassed` - check
`!completed, bypassed` - no check
`completed, bypassed` - don't care, either is fine

using isCompliant here would check in the third case, which I don't want. For this reason, it's sufficient to just look at `completed` to decide whether to check gsuite, which makes sense given my original comment in the code here (we're trying to track the actual 2FA status for the user in gsuite - bypass is irrelevant).